tests/devices/test_default_qutrit_mixed.py

```diff
@@ -1270,3 +1270,305 @@ def qfunc(x, y):
+
+        res = circuit()
+        assert np.allclose(res, expected)
+
+    # Expected expval list from circuit with diagonal observables after the measurement errors
+    # defined by gammas_and_probs are applied.
+    expected_commuting_expvals = [
+        [1 / 6, 1 / (2 * np.sqrt(3)), 1 / 6],
+        [-1 / 6, 1 / (2 * np.sqrt(3)), -1 / 6],
+        [-1 / 6, -1 / (2 * np.sqrt(3)), -1 / 6],
+        [1 / 3, 0, 1 / 3],
+        [5 / 6, 1 / (2 * np.sqrt(3)), 5 / 6],
+        [1 / 3, 1 / np.sqrt(3), 1 / 3],
+        [2 / 3, 1 / np.sqrt(3), 2 / 3],
+        [4 / 6, 0, 4 / 6],
+        [2 / 15, 1 / (10 * np.sqrt(3)), 2 / 15],
+        [4 / 15, 7 / (10 * np.sqrt(3)), 4 / 15],
+        [9 / 40, 3 / (40 * np.sqrt(3)), 9 / 40],
+    ]
+
+    @pytest.mark.parametrize(
+        "gammas_and_probs, expected", zip(gammas_and_probs, expected_commuting_expvals)
+    )
+    def test_readout_expval_commuting(self, nr_wires, gammas_and_probs, expected):
+        """Tests the measurement results for expval of GellMann 3 observables"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            wires=nr_wires,
+            damping_measurement_gammas=gammas_and_probs[0],
+            trit_flip_measurement_probs=gammas_and_probs[1],
+        )
+
+        @qml.qnode(dev)
+        def circuit():
+            self.setup_state(nr_wires)
+            return (
+                qml.expval(qml.GellMann(0, 3)),
+                qml.expval(qml.GellMann(0, 8)),
+                qml.expval(qml.GellMann(1, 3)),
+            )
+
+        res = circuit()
+        assert np.allclose(res, expected)
+
+    # Expected expval list from circuit with non-diagonal observables after the measurement errors
+    # defined by gammas_and_probs are applied. Gates are applied to the previous circuit so that
+    # the pre measurement probs are the same.
+    expected_noncommuting_expvals = [
+        [-1 / 3, -7 / 6, 1 / 6],
+        [-1 / 6, -1, -1 / 6],
+        [1 / 3, -1 / 6, -1 / 6],
+        [-1 / 6, -5 / 6, 1 / 3],
+        [-2 / 3, -3 / 2, 5 / 6],
+        [-2 / 3, -5 / 3, 1 / 3],
+        [-5 / 6, -11 / 6, 2 / 3],
+        [-1 / 3, -1, 4 / 6],
+        [-7 / 60, -49 / 60, 2 / 15],
+        [-29 / 60, -83 / 60, 4 / 15],
+        [-3 / 20, -101 / 120, 9 / 40],
+    ]
+
+    @pytest.mark.parametrize(
+        "gammas_and_probs, expected", zip(gammas_and_probs, expected_noncommuting_expvals)
+    )
+    def test_readout_expval_non_commuting(self, nr_wires, gammas_and_probs, expected):
+        """Tests the measurement results for expval of GellMann 1 observables"""
+        dev = qml.device(
+            "default.qutrit.mixed",
+            wires=nr_wires,
+            damping_measurement_gammas=gammas_and_probs[0],
+            trit_flip_measurement_probs=gammas_and_probs[1],
+        )
+        # Create matricies for the observables with diagonalizing matrix :math:`THadamard^\dag`
+        inv_sqrt_3_i = 1j / np.sqrt(3)
+        non_commuting_obs_one = np.array(
```

Comment:
I defined these matrices so that this test could be similar to the default_mixed readout error tests. One issue is that the eigenvalues/ diagonalizing_gates are ordered by the eigenvalues value. This was not what I had originally intended as I wanted to use the equivalent to GellMann_3 and GellMann_8 "undiagonalized" by adjoint(THadamard), this didn't work as GellMann_3's eigenvalues were reordered to [-1,0,1] which effectively reordered the states, changing their probs. Whats worse is since GellMann_8 has degeneracy the diagonalizing gates returned a linear combination of the first two states, making it hard to calculate. 

I instead used diag(-2,-1,1) which again is "undiagonalized" by adjoint(THadamard).
Since the eigenvalues are originally ordered I could just add up the probabilities times each eigenvalue for the expected value.

Reply:
Ah... is this because `eigh` is used internally for the eigenvalue calculation? I think I need to look at this again later (maybe during our 1:1) to fully parse what's happening.

Reply:
Just to clarify - the technique here is purely for the tests, correct? Could a user run into something unexpected because of this effect?

Reply:
The only way I think a user would really run into this is if they were to have a final diagonal matrix they expected and then wanted to add an "undiagonalizing" gate and input the matrix as a THermitian. 

I think it is fairly natural though as it doesn't give the wrong answer and it is likely what you would get if you put it into a QC unless you specifically set the eigenvalue order of your observable